docker/daemon.go

```diff
@@ -263,7 +264,21 @@ func (cli *DaemonCli) CmdDaemon(args ...string) error {
 	cli.TrustKeyPath = commonFlags.TrustKey
 
 	registryService := registry.NewService(cli.registryOptions)
-	d, err := daemon.NewDaemon(cli.Config, registryService)
+
+	remoteOpt := []libcontainerd.RemoteOption{
+		libcontainerd.WithDebugLog(cli.Config.Debug),
+	}
+	if cli.Config.ContainerdAddr != "" {
+		remoteOpt = append(remoteOpt, libcontainerd.WithRemoteAddr(cli.Config.ContainerdAddr))
+	} else {
+		remoteOpt = append(remoteOpt, libcontainerd.WithStartDaemon(true))
+	}
+	containerdRemote, err := libcontainerd.New(filepath.Join(cli.Config.ExecRoot, "libcontainerd"), remoteOpt...)
+	if err != nil {
+		logrus.Error(err)
```

Comment:
should we return if the error is not `nil`? if the error is not `nil`, the `containerdRemote` will nil, this will make the following `d, err := daemon.NewDaemon(cli.Config, registryService, containerdRemote)` failed with crash, because `d.containerd, err = containerdRemote.Client(d)` in `NewDaemon` will failed with `panic: runtime error: invalid memory address or nil pointer dereference`. A easy way to reproduce this is not install `containerd` in `$PATH`